napari/layers/vectors/vectors.py

```diff
@@ -773,15 +797,120 @@ def _get_value(self, position):
-        self._displayed_stored = copy(self._dims_displayed)
+
+    def _get_slice(self, request: _VectorSliceRequest) -> _VectorSliceResponse:
+        """New method"""
+        LOGGER.debug('Vectors._get_slice : %s', request)
+
+        # expand indices to include out of slice as needed,
+        # and get alpha values
+        slice_indices = self._get_slice_indices(request)
+        indices, alphas = self._slice_data(slice_indices, request.dims_not_displayed, request.out_of_slice_display)
+
+        # assume the mesh is up to date at this point -
+        # all mesh generation should be triggered by the new method `_update_mesh`
+
+        disp = list(request.dims_displayed)
+
+        # if there is no data, set empty
+        if len(request.data) == 0 or len(indices) == 0:
+            view_faces = np.array([[0, 1, 2]])
+            view_data = np.empty((0, 2, 2))
+            view_indices = []
+            view_vertices = np.zeros((3, request.ndisplay))
+        # if more than 2D, reduce the data and mesh dimensions
+        elif request.ndim > 2:
+            # TODO: is this line necessary?
+            indices, alphas = self._slice_data(slice_indices, request.dims_not_displayed, request.out_of_slice_display)
+
+            view_indices = indices
+            view_alphas = alphas
+            view_data = request.data[np.ix_(indices, [0, 1], disp)]
+            keep_inds = np.repeat(2 * indices, 2)
+            keep_inds[1::2] = keep_inds[1::2] + 1
+            if request.ndisplay == 3:
+                keep_inds = np.concatenate(
+                    [
+                        keep_inds,
+                        len(request.mesh_triangles) // 2 + keep_inds,
+                    ],
+                    axis=0,
+                )
+            view_faces = request.mesh_triangles[keep_inds]
+            view_vertices = request.mesh_vertices
+        else:
+            view_faces = request.mesh_triangles
+            view_vertices = request.mesh_vertices
+            view_data = request.data[:, :, disp]
+            view_indices = np.arange(request.data.shape[0])
+            view_alphas = 1.0
+
+        thumbnail = self._make_thumbnail(view_data, view_indices)
+
+        # prep for vispy by translating [z,y,x]->[x,y,z]
+        vertices = view_vertices[:, ::-1]
+
+        face_color = self._view_face_color(view_indices, view_alphas, request.ndisplay, request.ndim, request.edge_color)
+
+        if self._ndisplay == 3 and self.ndim == 2:
+            vertices = np.pad(vertices, ((0, 0), (0, 1)), mode='constant')
+
+        return _VectorSliceResponse(
+            request=request,
+            data=request.data,
+            data_to_world=request.data_to_world,
+            thumbnail=thumbnail,
+            view_data=view_data,
+            view_indices=view_indices,
+            view_faces=view_faces,
+            view_alphas=view_alphas,
+            view_vertices=vertices,
+            view_face_color=face_color,
```

Comment:
Nit: I think the `view_` prefix is unnecessary here because the response's main payload is the slice/view.

Reply:
Ha. I started out down that path, then turned back at some point to maintain naming consistency. I agree the `view_` is redundant. I'll fix.

Reply:
Done.